### ISSUE TITLE
Observatory is inaccessible (Missing or invalid authentication code)

### ISSUE THREAD

**Issue by miyoyo** [Dart-Code/Dart-Code#1729]
Haven't used the observatory for a while, but today I tried to open it, and it seems inaccessible.

![image](https://user-images.githubusercontent.com/3525610/58053419-c5a7f200-7b57-11e9-9604-ff0729298c2a.png)

Maybe #1721 is related?

```
[✓] Flutter (Channel beta, v1.5.4-hotfix.2, on Linux, locale en_US.UTF-8)
    • Flutter version 1.5.4-hotfix.2 at /opt/flutter
    • Framework revision 7a4c33425d (3 weeks ago), 2019-04-29 11:05:24 -0700
    • Engine revision 52c7a1e849
    • Dart version 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /opt/android-sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /opt/android-sdk
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b16-5323222)
    • All Android licenses accepted.

[✓] Android Studio (version 3.4)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 35.3.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b16-5323222)

[✓] Connected device (1 available)
    • Redmi 3S • c3fadefd7d33 • android-arm64 • Android 6.0.1 (API 23)

• No issues found!


```

[Dart-Code-Log-2019-04-01 23-36-01.txt](https://github.com/Dart-Code/Dart-Code/files/3199913/Dart-Code-Log-2019-04-01.23-36-01.txt)

**Comment by DanTup** [comment]
@miyoyo how did you launch Observatory?

**Comment by miyoyo** [comment]
Just via the palette's `Open Observatory"

Accidentally found out that adding a / at the end fixes it

**Comment by DanTup** [comment]
Yeah, I spotted the missing slash and that's why I wondered where it was coming from (when I tested it, there was a slash and it worked).

I'll figure out where it's being lost (or not being added). Thanks!

**Comment by DanTup** [comment]
Hmmm, I still can't repro. Could you run **Dart: Capture Logs** then launch your application, then when it's finished click **Stop Logging** and provide the log?

Thanks!

**Comment by miyoyo** [comment]
The log attached at the end of my original post is exactly that (I also open the observatory if that matters)

**Comment by DanTup** [comment]
@miyoyo Sorry, I just edited my comment - you need to include **Flutter Run** in the log, and have it running while the application launches (that's when we get the URL from Flutter). Thanks!

**Comment by miyoyo** [comment]
Woops, accidentally uploaded the wrong log, let me try again

**Comment by DanTup** [comment]
Oh, I think I know what's going on... It looks like we're losing the slash, but I think in a newer version of the VM it's doing a redirect when the slash is missing! 🙃

I'll fix it to preserve the slash in the next release anyway. Thanks for the info!

**Comment by miyoyo** [comment]
[Dart-Code-Log-2019-04-01 23-36-29.txt](https://github.com/Dart-Code/Dart-Code/files/3202176/Dart-Code-Log-2019-04-01.23-36-29.txt)
Much better

**Comment by DanTup** [comment]
@j0m3 please open a new issue with more details about your problem. The issue described here was fixed a long time ago, and there's not enough info on your comment to understand what's going on. Please include detail instructions of what you're doing leading up to this message and where you're seeing it (a screenshot could be useful). Thanks!
